### PR TITLE
BE-460, BE-461: Add user deletion endpoint to admin API

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Adds a user deletion endpoint to the Graph admin API. When invoked, it purges all entities in the user's personal web, deletes their Kratos identity (PII removal), revokes Hydra OAuth sessions, and removes email subscriptions — while preserving principals and policies for structural integrity.

Also makes entity deletion temporal axes configurable, adds the `LinkDeletionBehavior::Archive` variant for archiving incoming links during purge, and significantly expands deletion test coverage.

## 🔗 Related links

- [BE-460](https://linear.app/hash/issue/BE-460) _(internal)_
- [BE-311: Implement deletion for Entities](https://linear.app/hash/issue/BE-311) _(internal)_
- [BE-466: Erase scope does not clean up orphaned tombstones from prior Purge operations](https://linear.app/hash/issue/BE-466) _(internal)_

### User deletion orchestration
- New `POST /users/delete` admin endpoint accepting `{"userId": "..."}` or `{"email": "..."}`
- Orchestration in `hash-graph-store::user_deletion::delete_user()` — generic over store + external service traits
- Fatal steps (user lookup, entity purge) cause `Err` return; non-fatal steps (Kratos, Hydra, Mailchimp) collected via `ReportSink` into `UserDeletionOutcome::errors` with full error-stack context
- HTTP response message differentiates between full success and partial failure

### External service integrations
- `IdentityProvider` trait + `KratosIdentityProvider` — deletes Kratos identity (PII removal)
- `OAuthProvider` trait + `HydraOAuthProvider` — revokes login and consent sessions
- `EmailSubscriptionProvider` trait + `MailchimpSubscriptionProvider` — permanently deletes subscribers via Mailchimp Marketing API

### Entity deletion enhancements
- `DeleteEntitiesParams` now accepts `QueryTemporalAxesUnresolved` for configurable entity finding
- `LinkDeletionBehavior::Archive` — temporally archives incoming link entities before purge instead of rejecting or ignoring them. Uses an atomic CTE that closes `decision_time`, creates a historical row, and sets `archivedById` provenance
- `FOR UPDATE` locking on `entity_ids` during erase to close the TOCTOU gap with concurrent link creation
- New `AccountStore` methods: `get_user_kratos_identity_id`, `get_user_emails`, `get_user_id_by_email`

### Configuration
- Mailchimp clap `requires` constraints are now bidirectional (api_key requires list_id)
- External services config (Kratos URL, Hydra URL, Mailchimp) wired into admin server

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are in a state where docs changes are not _yet_ required but will be
  - this is tracked in: [BE-463: Write runbook for user account deletion](https://linear.app/hash/issue/BE-463)

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

## ⚠️ Known issues

- **Orphaned tombstones after Purge+Erase** (BE-466): `entity_ids` tombstones from prior Purge operations are invisible to subsequent Erase because `SelectCompiler` requires `entity_temporal_metadata` rows. A regression test exists but is `#[ignore]`d.

## 🐾 Next steps

- BE-463: Write operational runbook for user deletion
- BE-466: Fix orphaned tombstone cleanup
- Unit tests for `delete_user()` orchestrator with mocked dependencies

## 🛡 What tests cover this?

- 65 Rust integration tests in `libs/@local/graph/postgres-store/tests/deletion/` covering purge, erase, archive link behavior, draft handling, temporal axes validation, cross-web batches, and edge cases (self-loops, chains, idempotency)
- Backend integration tests in `user.test.ts` covering end-to-end user deletion (by ID and email), org entity survival, link archival provenance, and Kratos identity deletion
- Admin server tests in `admin-server.ts` for the `deleteUser` and `resetGraph` helpers
- HTTP test in `reset-database.http` updated with temporal axes

## ❓ How to test this?

1. Start the admin server: `cargo run --bin hash-graph --all-features -- admin-server --kratos-admin-url http://localhost:4434 --hydra-admin-url http://localhost:4445`
2. Create a test user via the regular API
3. Delete via admin API:
   ```bash
   curl -X POST http://localhost:4001/users/delete \
     -H "Content-Type: application/json" \
     -H "X-Authenticated-User-Actor-Id: 00000000-0000-0000-0000-000000000000" \
     -d '{"email": "user@example.com"}'
   ```
4. Verify the response includes `UserDeletionReport` with entity counts and service status booleans
5. Run Rust tests: `cargo nextest run --package hash-graph-postgres-store --test deletion`

### CODE REVIEW

File: libs/@local/graph/postgres-store/src/store/postgres/mod.rs [hunk 1]
```diff
@@ -4062,6 +4149,19 @@ where
             ))
             .await
             .change_context(DeletionError)?;
+        // Remove leftover entity_ids tombstones (from Purge scope) before deleting webs,
+        // because entity_ids has a FK on web.
+        self.as_client()
+            .client()
+            .simple_query("DELETE FROM entity_ids;")
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply libs/@local/graph/postgres-store/src/store/postgres/mod.rs:4164]: This deletes all principals, this means all webs should be gone, If there are still webs around, these are tombstones and needs to be removed as well. This is a test-endpoint to clean up the graph after tests. See BE-466.

File: tests/graph/benches/graph/scenario/stages/reset_db.rs [hunk 2]
```diff
@@ -89,6 +90,7 @@ impl ResetDbStage {
                         ActorEntityUuid::new(uuid::Uuid::nil()).into(),
                         DeleteEntitiesParams {
                             filter: Filter::All(Vec::new()),
+                            temporal_axes: QueryTemporalAxesUnresolved::default(),
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply tests/graph/benches/graph/scenario/stages/reset_db.rs:93]: `default()` resolves to the full axes. (I appreciate we may want to remove the `Default` implementation, but that's not a now-issue). See BE-467

File: apps/hash-graph/src/subcommand/admin_server.rs [hunk 3]
```diff
@@ -137,6 +137,32 @@ pub struct JwtConfig {
...
+
+    /// Mailchimp API key for email subscription management.
+    ///
+    /// The server prefix is extracted from the key (format: `<key>-<server>`, e.g. `abc123-us15`).
+    #[clap(long, env = "MAILCHIMP_API_KEY", requires = "mailchimp_list_id")]
+    pub mailchimp_api_key: Option<String>,
+
+    /// Mailchimp audience list ID.
+    #[clap(long, env = "MAILCHIMP_LIST_ID", requires = "mailchimp_api_key")]
+    pub mailchimp_list_id: Option<String>,
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-graph/src/subcommand/admin_server.rs:163]: That's expected and fine.

File: libs/@local/graph/postgres-store/src/store/postgres/knowledge/entity/delete.rs [hunk 4]
```diff
@@ -539,46 +536,221 @@ impl PostgresStore<Transaction<'_>> {
...
+                    FROM closed
+                ),
+                _provenance AS (
+                    UPDATE entity_editions
+                    SET provenance = provenance
+                        || jsonb_build_object('archivedById', $5::UUID)
+                    WHERE entity_edition_id IN (
+                        SELECT entity_edition_id FROM closed
+                    )
+                )
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply libs/@local/graph/postgres-store/src/store/postgres/knowledge/entity/delete.rs:615]: Yes, this is expected behavior. This is how a closed temporal bound looks like. The `archive_entity` implementation works the same way.

File: tests/hash-backend-integration/src/tests/graph/knowledge/system-types/user.test.ts [hunk 5]
```diff
@@ -321,12 +347,159 @@ describe("User model class", () => {
...
+      await expect(
+        kratosIdentityApi.getIdentity({
+          id: incompleteUser.kratosIdentityId,
+        }),
+      ).rejects.toThrow();
+    });
+  });
+
+  afterAll(async () => {
+    await resetGraph();
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply tests/hash-backend-integration/src/tests/graph/knowledge/system-types/user.test.ts:503]: if it fails the tests don't pass, right?
